bentoml/cli/deployment.py

```diff
@@ -213,3 +227,225 @@ def check_deployment_status(archive_path, platform, region, stage, api_name):
         deployment.check_status()
 
     return cli
+
+
+def parse_key_value_string(key_value_string):
+    if key_value_string:
+        result = {}
+        for item in key_value_string.split(','):
+            if item is not None:
+                splits = item.split('=')
+                result[splits[0]] = splits[1]
+        return result
+    else:
+        return None
+
+
+def get_deployment_operator_type(platform):
+    return DeploymentOperator.Value(platform.upper())
+
+
+def display_response_status_error(status):
+    _echo('Error code: {}'.format(status.status_code), CLI_COLOR_ERROR)
+    _echo('Error message: {}'.format(status.error_message), CLI_COLOR_ERROR)
+
+
+def display_deployment_info(deployment):
+    _echo('Deployment {} info\n\n'.format(deployment.name))
+    if deployment.namespace:
+        _echo('  namespace: {}'.format(deployment.namespace))
+    if deployment.labels:
+        _echo('  labels:')
+        for key, value in deployment.labels:
+            _echo('    - {key}: {value}'.format(key=key, value=value))
+    if deployment.annotations:
+        _echo('  annotations:')
+        for key, value in deployment.annotations:
+            _echo('    - {key}: {value}'.format(key=key, value=value))
+    if deployment.spec:
+        _echo('  Spec:')
+        _echo(
+            '    Bento: {name}:{version}'.format(
+                name=deployment.spec.bento_name, version=deployment.spec.bento_version
+            )
+        )
+    if deployment.state:
+        _echo('  State:')
+        _echo('    Current State: {}'.format(deployment.state.state))
+        if deployment.state.error_message:
+            _echo(
+                '    Error: {}'.format(deployment.state.error_message), CLI_COLOR_ERROR
+            )
+        _echo('    Info: {}'.format(deployment.state.info_json))
+
+
+def parse_bento_tag(tag):
+    if ':' in tag:
+        items = tag.split(':')
+        return items[0], items[1]
+    else:
+        return tag, 'latest'
+
+
+def get_deployment_sub_command(cli):
+    @click.group()
+    def deploy():
+        pass
+
+    @deploy.command(
+        short_help="Apply deployment configuration",
+        context_settings=dict(ignore_unknown_options=True, allow_extra_args=True),
+    )
+    @click.argument('--bento-tag', type=click.STRING, required=True)
+    @click.argument("--deployment-name", type=click.STRING, required=True)
+    @click.option(
+        "--platform",
+        type=click.Choice(
+            ["aws_lambda", "gcp_function", "aws_sagemaker", "kubernetes", "custom"]
+        ),
+        required=True,
+        help="Target platform that Bento archive is going to deployed to",
+    )
+    @click.option("--namespace", type=click.STRING, help="Deployment's namespace")
+    @click.option("--labels", type=click.STRING)
```

Comment:
add help message, and examples? something like '--labels=foo=123,env=staging'

Reply:
make sure to denote that if an option is only effective for certain deploy platform